platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListOffHeap.java

```diff
@@ -129,11 +129,18 @@ public LongListOffHeap(final int reservedBufferLength) {
      */
     public LongListOffHeap(final Path file) throws IOException {
         super(FileChannel.open(file, StandardOpenOption.READ));
-        final int numberOfChunksWithData = calculateNumberOfChunks(size());
+        final int totalNumberOfChunks = calculateNumberOfChunks(size());
+        final int firstChunkWithDataIndex = (int) (minValidIndex.get() / numLongsPerChunk);
+        final int firstValidIndex = (int) (minValidIndex.get() % numLongsPerChunk);
         chunkList = new AtomicReferenceArray<>(calculateNumberOfChunks(maxLongs));
         reservedBufferLength = DEFAULT_RESERVED_BUFFER_LENGTH;
-        // read data
-        for (int i = 0; i < numberOfChunksWithData; i++) {
+        // read the first chunk
+        final ByteBuffer firstBuffer = createChunk();
+        firstBuffer.position(firstValidIndex * Long.BYTES).limit(firstBuffer.capacity());
+        MerkleDbFileUtils.completelyRead(fileChannel, firstBuffer);
+        chunkList.set(firstChunkWithDataIndex, firstBuffer);
+        // read the rest of the data
+        for (int i = firstChunkWithDataIndex + 1; i < totalNumberOfChunks; i++) {
             final ByteBuffer directBuffer = createChunk();
             MerkleDbFileUtils.completelyRead(fileChannel, directBuffer);
```

Comment:
The last chunk may be shorter that numLongsPerChunk. I hope MerkleDbFileUtils.completelyRead() can handle that just fine

Reply:
Yes, you're right - `MerkleDbFileUtils.completelyRead()` handles this case: 

```
        while (dstBuffer.hasRemaining()) {
            final long bytesRead = fileChannel.read(dstBuffer);
            if (bytesRead < 0) {
                // Reached EOF
                break;
            }
            totalBytesRead += bytesRead;
```
If file channel has no more data it stops reading.